### ISSUE TITLE
使用已训练好的URanker网络，不需要with torch.no_grad() ？

### ISSUE THREAD

**Issue by Ynqqq2022** [RQ-Wu/UnderwaterRanker#2]
您好，感谢您出色的工作！我有一个疑问，在使用已训练好的URanker网络计算损失函数时，为什么不需要加with torch.no_grad()？这样optimizer.step()岂不是会更新URanker中的参数吗？如果提了愚蠢的问题，还请见谅！https://github.com/RQ-Wu/UnderwaterRanker/blob/8e1928bb85583f0a731ed63040266a629e6c108f/loss.py#L16-L22

**Comment by RQ-Wu** [comment]
只有NU2Net放到了optimizer里面，所以只会更新增强网络的参数。加了torch.no_grad后loss的梯度就传不过去了，同理VGG loss也没有加torch.no_grad，这个是我之前做实验的时候想省显存犯的一个错误，然后我给注释掉了。

**Comment by Ynqqq2022** [comment]
明白了，万分感谢！
